addons/mail/static/tests/owl/component/discuss_tests.js

```diff
@@ -3739,6 +3743,115 @@ QUnit.test('composer: add an attachment', async function (assert) {
         "should have an attachment remove button");
 });
 
+QUnit.test('composer: drop attachments', async function (assert) {
+    assert.expect(3);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: 'channel',
+                id: 20,
+                name: "General",
+            }],
+        },
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_fetch') {
+                return [];
+            }
+            return this._super(...arguments);
+        },
+        params: {
+            default_active_id: 'mail.channel_20',
+        },
+    });
+    const files = [
+        await testUtils.file.createFile({
+            name: 'text.txt',
+            content: 'hello, world',
+            contentType: 'text/plain',
+        }),
+        await testUtils.file.createFile({
+            name: 'text2.txt',
+            content: 'hello, worlduh',
+            contentType: 'text/plain',
+        })
+    ];
+
+    await dragoverFiles(document.querySelector('.o_Thread'));
+    assert.ok(
+        document.querySelector('.o_Thread_dropZone'),
+        "should have a drop zone");
+
+    await dropFiles(
+        document.querySelector('.o_Thread_dropZone'),
+        files);
+
+    assert.ok(
+        document.querySelector('.o_Composer_attachmentList'),
+        "should have an attachment list");
+    assert.strictEqual(
+        document
+            .querySelectorAll(`
+            .o_Composer_attachmentList
+            .o_Attachment`)
+            .length,
+        2,
+        "should have two attachments");
```

Comment:
- should have 2 attachments in the composer
- test should check that attachment are what we expect, maybe order of attachments should be checked?